spec/grover/middleware_spec.rb

```diff
@@ -860,6 +860,46 @@
       end
     end
 
+    describe '#ignore_request?' do
+      context 'when app configuration has a Proc for ignore_request' do
+        context 'with a hostname' do
+          before { allow(Grover.configuration).to receive(:ignore_request).and_return(->(req) { req.host == 'www.example.org' }) }
+
+          it 'request is ignored when the request path passed to the proc defined in ignore_request returns true' do
+            response = get 'http://www.example.org/foobazbar'
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobazbar.pdf'
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.example.org/foobarbar'
+            expect(response.content_type).to eq 'text/html'
+
+            response = get 'http://www.therealexample.org/foobarbar.pdf'
```

Comment:
same applies to these tests. The first two are `www.example.org`, the second two should be something different (not just this one). And I would also suggest that the path (except the extension where appropriate) should be the same. ie `foobazbar[.pdf]`

Reply:
OK, I think I got you.

1. blocking with host (html request) returns html
2.blocking with host (pdf request) returns html
3. not blocking with host (html request) returns html
4. not blocking with host (pdf request) returns pdf